### ISSUE TITLE
vue-tsc cannot load plugin @vue-macros/volar/better-define Error

### ISSUE THREAD

**Issue by richardtallent-erm** [sxzz/unplugin-vue-macros#229]
### Describe the bug

When running `vue-tsc --noEmit --skipLibCheck`, I get the following errors:

```sh
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './better-define' is not 
defined by "exports" in C:\[PROJECT PATH]\node_modules\@vue-macros\volar\package.json
    at new NodeError (node:internal/errors:372:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:472:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:753:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.resolve (node:internal/modules/cjs/helpers:108:19)
    at C:\[PROJECT PATH]\node_modules\@volar\vue-language-core\out\utils\ts.js:50:30
    at Array.map (<anonymous>)
    at createParsedCommandLineBase (C:\[PROJECT PATH]\node_modules\@volar\vue-language-core\out\utils\ts.js:48:92) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
Load plugin failed @vue-macros/volar/better-define Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './better-define' is not defined by "exports" in C:\[PROJECT PATH]\node_modules\@vue-macros\volar\package.json
    at new NodeError (node:internal/errors:372:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:472:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:753:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.resolve (node:internal/modules/cjs/helpers:108:19)
    at getDefaultVueLanguagePlugins (C:\[PROJECT PATH]\node_modules\@volar\vue-language-core\out\plugins.js:30:44)
    at Object.createLanguageModules (C:\[PROJECT PATH]\node_modules\@volar\vue-language-core\out\languageModule.js:9:74)
    at Object.createLanguageService (C:\[PROJECT PATH]\node_modules\@volar\vue-typescript\out\index.js:6:66) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

The command runs successfully, I just get the errors above. I get the same errors with Volar in VS Code in the "Vue Semantic Server" Output tab.

I just started using `vue-tsc` from the command line today and noticed this, so it may not be a recent issue.

### Reproduction

https://github.com/richardtallent

### System Info
Apologies, I don't have time at the moment to work up a minimal example. I'm hoping this is some minor obvious issue with the packaging process, i.e., not anything that would be specific to my tsconfig or other repo-specific issues. But if not and if a repo is needed, please let me know and I'll follow up as soon as I can.

- Vue: 3.2.45
- Vite: 4.0.4
- unplugin-vue-macros: ^1.4.2

```Shell

  System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 15.47 GB / 31.73 GB
  Binaries:
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
    npm: 8.13.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (109.0.1518.61)
    Internet Explorer: 11.0.19041.1566
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sxzz** [comment]
Please try the latest version and provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue. Thanks.

https://antfu.me/posts/why-reproductions-are-required

**Comment by richardtallent** [comment]
Latest version didn't resolve the issue, unfortunately. Minimal example here:

https://github.com/richardtallent/unplugin-vue-macros-tsc-error

To see the TypeScript error:

```Shell
git clone https://github.com/richardtallent/unplugin-vue-macros-tsc-error.git
cd unplugin-vue-macros-tsc-error
npm install
npm run tsc
```

Thanks!

**Comment by sxzz** [comment]
Oh, I just realized that you just used `@vue-macros/volar/better-define` as a volar plugin. Actually, betterDefine doesn't need a volar plugin. I'll improve the docs later. Thanks for your reproduction!

**Comment by richardtallent-erm** [comment]
Thanks! I removed the volar plugin (I'm only using better-define currently) and tsc works now.

But when upgraded to 1.6.1, now I'm getting this (with both my app and the sample repo above). Should I submit a separate issue?

```
vite build

failed to load config from C:\Temp\VS\unplugin-vue-macros-tsc-error\vite.config.ts
error during build:
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in C:\Temp\VS\unplugin-vue-macros-tsc-error\node_modules\@vue-macros\reactivity-transform\node_modules\estree-walker\package.json
    at new NodeError (node:internal/errors:372:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:472:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:693:7)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Temp\VS\unplugin-vue-macros-tsc-error\node_modules\@vue-macros\reactivity-transform\dist\chunk-IP5BVLGV.js:14:21)
```

**Comment by sxzz** [comment]
@richardtallent-erm Fixed via 5fee1ad. You can try the next version later.
